apps/web/src/components/providers/DocsProvider.tsx

```diff
@@ -0,0 +1,192 @@
+import { useSegment } from '@novu/shared-web';
+
+const Text = styled('p', text);
+
+interface IDocsContext {
+  setPath: (path: string) => void;
+  toggle: () => void;
+  enabled: boolean;
+  path: string;
+}
+
+const DocsContext = React.createContext<IDocsContext>({
+  setPath: () => {},
+  toggle: () => {},
+  enabled: false,
+  path: '',
+});
+
+export const useDocsContext = (): IDocsContext => useContext(DocsContext);
+
+export const useSetDocs = (path: string) => {
+  const { setPath } = useDocsContext();
+  useEffect(() => {
+    setPath(path);
+  }, [path, setPath]);
+};
+
+export const DocsProvider = ({ children }) => {
+  const [path, setPath] = useState<string>('');
+  const [open, setOpen] = useState(false);
+  const [voted, setVoted] = useState('');
+  const { pathname } = useLocation();
+  const segment = useSegment();
+  const { colorScheme } = useMantineColorScheme();
+  const isDark = colorScheme === 'dark';
+
+  const onClose = () => {
+    toggle();
+  };
+
+  useEffect(() => {
+    return () => {
+      setPath('');
+      setVoted('');
+    };
+  }, [pathname]);
+
+  const toggle = () => {
+    setOpen(!open);
+  };
+
+  const onVoteClick = (vote: string) => () => {
+    if (voted.length > 0) {
+      return;
+    }
+    segment.track('Inline docs voting used', {
+      documentationPage: path,
+      pageURL: window.location.href,
+      vote,
+    });
+    setVoted(vote);
+  };
+
+  return (
+    <DocsContext.Provider
+      value={{
+        setPath,
+        toggle,
+        enabled: path.length > 0,
+        path,
+      }}
+    >
+      <Modal
+        opened={open}
+        onClose={onClose}
+        styles={{
+          root: {
+            zIndex: 10003,
+          },
+          inner: {
+            padding: '24px',
+          },
+          body: {
+            height: '100%',
+            maxHeight: '100%',
+          },
+          modal: {
+            width: 800,
+            padding: '24px !important',
+            borderRadius: 12,
+            position: 'relative',
+            height: '100%',
+          },
+        }}
+        overflow="inside"
+        withCloseButton={false}
+        overlayColor={isDark ? colors.BGDark : colors.BGLight}
+      >
+        <Docs path={path}>
+          <Flex
+            className={css({
+              position: 'fixed',
+              top: '150',
+              right: '150',
+              background: isDark ? 'legacy.B15' : 'white',
+              zIndex: 1,
+              padding: '25',
+              borderBottomLeftRadius: '50',
+            })}
+            gap="125"
+          >
+            <Tooltip label="Open docs website">
+              <ActionIcon
+                style={{
+                  width: '20px  !important',
+                  minWidth: '20px !important',
+                  border: 'none',
+                }}
+                variant="transparent"
+                onClick={() => {
+                  window.open(`https://docs.novu.co/${path}`);
```

Comment:
```suggestion
                  window.open(`https://docs.novu.co/${path}`, '_blank', 'noopener,noreferrer,resizable');
```
Please use env var.

Reply:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Should we open to a new tab or window?‏

Reply:
Window.open already opens this in a new tab as seen in the video :)